src/error_chain.rs

```diff
@@ -137,6 +137,17 @@ macro_rules! error_chain_processed {
                 )
             }
 
+            /// Construct a chained error from another boxed error and a kind, and generates a backtrace
+            pub fn with_boxed_chain<K>(error: Box<::std::error::Error + Send>, kind: K)
+                -> $error_name
+                where K: Into<$error_kind_name>
+            {
+                $error_name(
+                    kind.into(),
+                    $crate::State::new::<$error_name>(error, ),
+                )
+            }
+
```

Comment:
This is how I allow boxed errors to interact with error_chain.

It's not possible to define a separate `ResultExt` for results which contain boxed errors due to Rust's type inference--it won't allow for 2 traits with the same concrete types & type variables but with different type constraints (This rules out an approach as in the comment).

I'm open to other approaches to allow for boxed errors to be introduced in the error-chain, but I think the only better approach would be to create another trait (for example `ResultExt2`), and have it also have `chain_err` and implement it for boxed error `Result` types. I don't want to implement that, however, because it's a large change to the signature of the macro (a new 5th type name is needed), and I'm not sure it won't just have different type inference issues due to the `chain_err` name matching.